### PR TITLE
LUI-167: Should be able to find a program workflow by id or uuid

### PR DESCRIPTION
**Pull request by rubailly**
- Added #`getProgramWorkflow` in `DWRProgramWorkflowService.java ` for retrieving a `ProgramWorkflow` by uuid or id

- Modified #`getStatesByWorkflow` to use #`getProgramWorkflow` for getting `ProgramWorkflow`s

### CODE REVIEW

**Comment by rubailly** [comment]
@mseaton @dkayiwa  I noticed a few failing unit tests after upgrading the OpenMRS core dependency to 2.3.2, I'll look into it.

**Comment by HerbertYiga** [comment]
@rubailly  Thanks for the works here, you forgot to add the ticket ID on your description

**Comment by dkayiwa** [comment]
@rubailly do you mind adding some tests here? DWRProgramWorkflowServiceTest

**Comment by rubailly** [comment]
@dkayiwa [Done](https://github.com/openmrs/openmrs-module-legacyui/pull/129/commits/4d7ef0ea145633e0fe33090bceaf9657e80cabb7).

**Comment by dkayiwa** [comment]
@rubailly can you also request for code review and include the pull request link on the JIRA ticket as recommended at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

File: omod/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java [hunk 1]
```diff
@@ -85,11 +85,50 @@ public PatientProgramItem getPatientProgram(Integer patientProgramId) {
...
+			}
+		}
+		return workflow;
+	}
+	
+	public Vector<ListItem> getStatesByWorkflow(String workflowLookup) {
+		log.debug("In getStatesByWorkflow with workflow uuid of " + workflowLookup);
 		Vector<ListItem> ret = new Vector<ListItem>();
 		
-		ProgramWorkflow workflow = Context.getProgramWorkflowService().getWorkflowByUuid(uuid);
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java:92]: What is the value of changing this from the service method call?

Comment (rubailly) [reply omod/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java:92]: [This](https://github.com/openmrs/openmrs-core/blob/1.9.11/web/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java#L86) is how the method was written in 1.9.x , it was expecting an Integer and that's how modules have been calling it. See example [here](https://github.com/openmrs/openmrs-module-reportingcompatibility/blob/635fe83e0ca793b1e0a3c0e7d3923f84c9e10b91/omod/src/main/webapp/analysis/cohortBuilder.jsp#L398).
Somehow, [it was changed](https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/src/main/java/org/openmrs/web/dwr/DWRProgramWorkflowService.java#L88) to now expect a String (UUID) and that means all modules that are passing an Integer are now failing. 
So, I'm replacing the service call (Context.getProgramWorkflowService().getWorkflowByUuid(uuid);) with a more generic method to make sure modules that use other parameter types (names, and Ids) don't fail.

**Pull request merged**